lib/dialects/abstract/query.js

```diff
@@ -478,6 +478,7 @@ class AbstractQuery {
 
       return uniqueKeyAttributes;
     };
+    const stringify = obj => Buffer.isBuffer(obj) ? obj.toString('hex') : String(obj);
```

Comment:
Did you benchmark this? This section of the code is highly performance sensitive.

Reply:
I haven't, but even if I would, I do not know different approach to solve this. Can you help?

Reply:
Something might be faster than than Buffer.isBuffer and i would definitely not waste time casting everything else to strings

Reply:
The method is likely so small it will be inlined, but it still does mean a Buffer.isBuffer call on each row (of which there can be a million easily)

Reply:
Check, i will remove casting to string everything else.  Is there anything in place for sequelize already?

Reply:
I've run simple benchmark of `isBuffer` and `instance of Buffer` locally a couple of times, using benchmark npm package:

```
Buffer.isBuffer x 53,576,349 ops/sec ±2.07% (87 runs sampled)
instanceof Buffer x 81,705,405 ops/sec ±2.34% (86 runs sampled)
```
It appears that instanceof is faster. Is this enough or do you want me to do some other benchmarking?